src/blosc2/proxy.py

```diff
@@ -296,6 +292,58 @@ async def afetch(self, item: slice | list[slice] = None) -> blosc2.NDArray | blo
         -----
         This method is only available if the :ref:`ProxySource` or :ref:`ProxyNDSource`
         have an async `aget_chunk` method.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import blosc2
+        >>> import asyncio
+        >>> class MyProxySource:
+        >>>     def __init__(self, data, chunk, block):
+        >>>             # If the next source is multidimensional, it must have the attributes:
+        >>>             self.data = data
+        >>>             self.shape = data.shape
+        >>>             self.chunks = chunk
+        >>>             self.blocks = block
+        >>>             self.dtype = data.dtype
+        >>>             f"Data shape: {self.shape}, Chunks: {self.chunks}"
+        Data shape: (4, 5), Chunks: [2, 2]
+        >>>             f"Blocks: {self.blocks}, Dtype: {self.dtype}"
+        Blocks: [2, 2], Dtype: int64
+        >>> # This method must be present.
+        >>>     async def aget_chunk(self, nchunk):
+        >>>             await asyncio.sleep(0.1)
+        >>>             return self.data[nchunk]
+        >>> class MyProxy(blosc2.Proxy):
+        >>>     def __init__(self, source):
+        >>>               super().__init__(source)
+        >>>               self._cache = source.data
+        >>>     # Asynchronous method to get the cache data
+        >>>     async def afetch(self, slice_=None):
+        >>>               return self._cache if slice_ is None else self._cache[slice_]
+        >>> data = np.arange(20).reshape(4, 5)
+        >>> chunk = [2, 2]
+        >>> block = [2, 2]
+        >>> source = MyProxySource(data, chunk, block)
+        >>> proxy = MyProxy(source)
+        >>> async def fetch_data():
+        >>>         # Fetch a slice of the data from the proxy asynchronously
+        >>>         slice_data = await proxy.afetch(slice(0, 2))
+        >>>         f"Slice data cache: {slice_data[:]}"
+        Slice data cache:
+                            [[0 1 2 3 4]
+                            [5 6 7 8 9]]
+        >>>         # Note: It is not possible to print the intermediate state of the proxy
+        >>>         # because accessing the proxy causes the entire data to be copied.
+        >>>         # Fetch the full data from the proxy asynchronously
+        >>>         full_data = await proxy.afetch()
```

Comment:
If you store the cache on disk like `proxy = blosc2.Proxy(source, urlpath="proxy_cache.b2nd")`, you can check that the cache is empty before accessing the data with 
```
cache = blosc2.open("proxy_cache.b2nd")
print(cache[slice]) # This shouldn't be initialized
print(proxy[slice]) # Access the data to initialize the cache
print(cache[slice]) # This should now be initialized
```